docs/src/reference/grid.md

```diff
@@ -2,4 +2,108 @@
 DocTestSetup = :(using Ferrite)
 ```
 
-# Grid
\ No newline at end of file
+# Grid & AbstractGrid
+
+## Grid
+
+```@docs
+Node
+Cell
+CellIndex
+VertexIndex
+EdgeIndex
+FaceIndex
+Grid
+```
+
+### Utility Functions
+
+```@docs
+getcells
+getncells
+getnodes
+getnnodes
+Ferrite.nnodes_per_cell
+getcellset
+getcellsets
+getnodeset
+getnodesets
+getfaceset
+getfacesets
+getedgeset
+getedgesets
+getvertexset
+getvertexsets
+compute_vertex_values
+transform!
+getcoordinates
+getcoordinates!
+```
+
+### Grid Sets Utility
+
+```@docs
+addcellset!
+addfaceset!
+addnodeset!
+```
+
+## AbstractGrid
+
+It can be very useful to use a grid type for a certain special case, e.g. mixed cell types, adaptivity, IGA, etc.
+In order to define your own `<: AbstractGrid` you need to fulfill the `AbstractGrid` interface.
+In case that certain structures are preserved from the `Ferrite.Grid` type, you don't need to dispatch on your own type, but rather rely on the fallback `AbstractGrid` dispatch.
```

Comment:
I thought it's worth mentioning because of two reasons: on one hand it's the only interface documentation Ferrite.jl has and on the other hand even when I started to subtype an AbstractGrid this wasn't really clear to me, because you need to dig in the source code which functions are defined based on `AbstractGrid` and use `Grid`'s structure and which dispatch only on `Grid` even though`AbstractGrid` could have been used there as well . But I don't mind deleting it, if you want to :)

Reply:
Sounds like good reasoning, let's leave it as it is then. :)